src/metabase/pulse/render/body.clj

```diff
@@ -528,37 +527,35 @@
       (h value)]
      :render/text (str value)}))
 
+(defn- result-k->data-k
+  [{:keys [result] :as m}]
+  (-> m
+      (assoc :data (:data result))
+      (dissoc :result))
+  #_(m/map-keys (fn [k] (if (#{:result} k) :data k)) m))
+
 ;; the `:javascript_visualization` render method
 ;; is and will continue to handle more and more 'isomorphic' chart types.
 ;; Isomorphic in this context just means the frontend Code is mostly shared between the app and the static-viz
 ;; As of 2024-03-21, isomorphic chart types include: line, area, bar (LAB), and trend charts
 ;; Because this effort began with LAB charts, this method is written to handle multi-series dashcards.
 ;; Trend charts were added more recently and will not have multi-series.
-;; Despite this, the function `pu/execute-multi-card` will still correctly execute dashcards.
-;; This conditional is here to cover the case of trend charts in Alerts (not subscriptions). Alerts
-;; exist on Questions and thus have no associated dashcard, which causes `pu/execute-multi-card` to fail.
 (mu/defmethod render :javascript_visualization :- formatter/RenderedPulseCard
   [_chart-type render-type _timezone-id card dashcard data]
-  (let [combined-cards-results (if dashcard
-                                 (pu/execute-multi-card card dashcard)
-                                 [(pu/execute-card {:creator_id (:creator_id card)} (:id card))])
-        cards-with-data        (m/distinct-by
-                                #(get-in % [:card :id])
-                                (map
-                                 (comp
-                                  add-dashcard-timeline-events
-                                  (fn [c d] {:card c :data d}))
-                                 (cond-> (map :card combined-cards-results)
-                                   dashcard (conj card))
-                                 (cond-> (map #(get-in % [:result :data]) combined-cards-results)
-                                   dashcard (conj data))))
-        dashcard-viz-settings  (get dashcard :visualization_settings)
-        {rendered-type :type content :content} (js-svg/javascript-visualization cards-with-data dashcard-viz-settings)]
+  (let [series-cards-results                   (:series-results dashcard)
+        cards-with-data                        (->> series-cards-results
+                                                    (map result-k->data-k)
```

Comment:
```suggestion
                                                    (map #(clojure.set/rename-keys % {:result :data}))
```
this should do the same thing

Reply:
Ah, turns out I do need to use a function here to achieve what I need: The :data key inside the result key of the card passed in is what we need, we don't want to just rename :result to :data because that results in this:
`{:data {:data {...}}` when what we actually want is: `{:data {...}}`